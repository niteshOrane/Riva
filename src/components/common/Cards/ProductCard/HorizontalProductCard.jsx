import React from "react";
import styles from "./horizontalProductCard.module.scss";

const VerticalProductCard = ({ product }) => {
  const { name, src, was, now, size } = product;

  return (
    <div className={`${styles.horizontalProductCard} d-flex gap-12px`}>
      <div>
        <img src={src} alt={name} />
      </div>
      <div>
        <div className={styles.name}>{name}</div>
        <div className={styles.price}>
          <div className={styles.was}>Was {was}$</div>
          <div className={styles.now}>Now {now}$</div>
        </div>
        <div className={styles.size}>
          <div className="gap-12px d-flex align-items-center">
            <span className={styles.title}>Size:</span>
            <span className={styles.text}>{size}</span>
          </div>
          <div
            className={`${styles.options} gap-12px d-flex align-items-center`}
          >
            {["S", "M", "L", "XL", "2XL"].map((sizeValue) => {
              return <div className={styles.option}>{sizeValue}</div>;
            })}
          </div>
          <div className={`${styles.otherOptions} d-flex`}>
            Other similar options
            <span className="material-icons-outlined font-light-black">
              chevron_right
            </span>
          </div>
        </div>
      </div>
    </div>
  );
};

export default VerticalProductCard;
