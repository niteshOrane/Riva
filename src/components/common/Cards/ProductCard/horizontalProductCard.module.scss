.horizontalProductCard {
    height: 150px;

    img {
        width: 100px;
    }

    .price {
        font-size: 10px;
        font-weight: 500;
        margin-top: 5px;
        justify-content: center;

        .was {
            display: inline-block;
            height: 18px;
            margin-right: 8px;
            font-size: 15px;
            font-weight: 700;
            font-stretch: normal;
            font-style: normal;
            line-height: 1.47;
            letter-spacing: normal;
            text-align: center;
            color: #5c5959;
            text-decoration: line-through;
        }

        .now {
            display: inline-block;
            height: 18px;
            font-size: 15px;
            font-weight: 700;
            margin-left: 5px;
            font-stretch: normal;
            font-style: normal;
            line-height: 1.47;
            text-align: center;
            letter-spacing: normal;
            color: #d12003;
        }
    }

    .size {
        margin-top: 10px;
        margin-bottom: 14px;
        .title {
            font-size: 14px;
            font-weight: 800;
            color: #161616;
        }

        .options {
            margin-top: 7px;
            .option {
                display: flex;
                align-items: center;
                justify-content: center;
                font-size: 14px;
                border: solid 1px #b2aeae;
                font-family: Roboto;
                text-align: center;
                width: 40.1px;
                height: 24px;
                cursor: pointer;
                &:hover{
                    background-color: #F5F5F5;
                }
            }
        }
    }

    .otherOptions{
        font-family: Roboto;
        font-size: 14px;
        font-weight: 500;
        font-stretch: normal;
        font-style: normal;
        line-height: 1.57;
        letter-spacing: normal;
        text-align: left;
        color: #161616;
        align-items: center;
        margin-top: 14px;
        margin-bottom: 25px;
        cursor: pointer;
    }
}

.text{
    cursor: pointer;
    white-space: nowrap;
    font-size: 14px;
    border: solid 1px #b2aeae;
    padding:3px 5px
}
