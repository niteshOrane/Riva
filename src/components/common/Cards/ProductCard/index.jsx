import React from 'react';
import PropTypes from 'prop-types';
import Image from '../../LazyImage/Image';
import styles from './product.module.scss';

const ProductCard = ({ product }) => {
  const { id, src, name, wasPrice, nowPrice, images } = product;
  return (
    <div key={id} className={styles.productCard}>
      <div className={styles.imageContainer}>
        <Image src={src || images[0]} />
      </div>
      <div className={styles.actionContainer}>
        <div>
          <span className="material-icons-outlined font-light-black">
            favorite_border
          </span>
        </div>
        <div>
          <span className="material-icons-outlined font-light-black">
            search
          </span>
        </div>
        <div>
          <span className="material-icons-outlined font-light-black">
            shopping_cart
          </span>
        </div>
      </div>
      <div className={styles.productName}>{name}</div>
      <div className={styles.productPrice}>
        <div className={styles.was}>Was {wasPrice}$</div>
        <div className={styles.now}>Now {nowPrice}$</div>
      </div>
      <div className={styles.productColors}>
        <div className={`${styles.color} ${styles.color_red}`} />
        <div className={`${styles.color} ${styles.color_oranage}`}/>
        <div className={`${styles.color} ${styles.color_blue}`}/>
      </div>
    </div>
  );
};
export default ProductCard;
