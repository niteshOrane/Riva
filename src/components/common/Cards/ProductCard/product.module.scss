.productCard {
  // max-width: 281px;
  // width: 281px;
  // height: 400px;
  width: 100%;
  margin: 10px 0px 10px 0px;
  text-align: center;
  font-family: arial;
  position: relative;
  display: inline-block;
  background-color: #ffffff;
  &:hover {
    box-shadow: 0.4px 5px 8px 0 rgba(212, 206, 206, 0.39);
  }

  .imageContainer img {
    position: relative;
    width: 100%;
    object-fit: contain;
    background: #ddd;
  }

  .productName {
    height: 18px;
    font-size: 15px;
    font-weight: 500;
    font-stretch: normal;
    font-style: normal;
    line-height: 1.47;
    letter-spacing: normal;
    text-align: center;
    color: #161616;
    cursor: pointer;
    margin-top: 5px;
  }

  .productPrice {
    font-size: 10px;
    font-weight: 500;
    margin-top: 5px;
    justify-content: center;

    .was {
      display: inline-block;
      height: 18px;
      margin-right: 8px;
      font-size: 15px;
      font-weight: normal;
      font-stretch: normal;
      font-style: normal;
      line-height: 1.47;
      letter-spacing: normal;
      text-align: center;
      color: #5c5959;
      text-decoration: line-through;
    }

    .now {
      display: inline-block;
      height: 18px;
      font-size: 15px;
      font-weight: normal;
      margin-left: 8px;
      font-stretch: normal;
      font-style: normal;
      line-height: 1.47;
      text-align: center;
      letter-spacing: normal;
      color: #d12003;
    }
  }

  .actionContainer {
    position: absolute;
    top: 4px;
    right: 4px;
    font-size: 15px;
    font-weight: 500;
    background-color: transparent;

    div {
      margin: 5px 5px 0px 0px;
      box-shadow: 0.3px 3px 2px 0 rgba(148, 144, 144, 0.33);
      background-color: #ffffff;
      border-radius: 50%;
      cursor: pointer;
      &:hover {
        transform: scale(1.1);
      }
    }

    span {
      padding: 5px;
      font-size: 15px;
    }
  }
}

.color {
  height: 13px;
  display: inline-block;
  width: 13px;
  margin: 2px;
  background-color: #b28965;
  border-radius: 50%;
  // box-shadow: 0px 0px 0px 1px #ffffff inset;
  &:hover {
    transform: scale(1.25);
    filter: brightness(80%);
  }
  &_red {
    background-color: #e45850;
  }
  &_blue {
    background-color: #23a7f3;
  }
  &_oranage {
    background-color: #f58c30;
  }
  cursor: pointer;
}
