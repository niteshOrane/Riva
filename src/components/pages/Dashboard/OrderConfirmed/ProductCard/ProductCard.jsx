import React from "react";
import styles from "./ProductCard.module.scss";
function ProductCard({ product }) {
  return (
    <div className={styles.card}>
      <div className={styles.image}>
        <img src={product?.src} width="100%" alt={product?.name} />
      </div>
      <div className={styles.productDetails}>
        <h4 className={styles.name}>{product?.name}</h4>
        <h4 className={styles.price}>{product?.price}</h4>
        <div className="d-flex align-items-center">
          <span className={`${styles.sizeColor} ${styles.mt4}`}>Color: </span>{" "}
          <span>{product?.color}</span>
        </div>
        <div className="d-flex align-items-center">
          <span className={styles.sizeColor}>Size: </span>{" "}
          <span>{product?.size}</span>
        </div>
        <p className={styles.mt4}>Order Placed @{product?.placedDate}</p>
        <p>Order ID #${product?.orderId}</p>
        <p>Payment {product?.payment}</p>
      </div>
    </div>
  );
}

export default ProductCard;
