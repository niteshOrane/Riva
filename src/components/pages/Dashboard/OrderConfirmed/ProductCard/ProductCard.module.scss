.card {
  padding: 12px;
  margin-top: 5px;
  border: solid 1px #c2bfbf;
  display: flex;
  height: fit-content;
  .image {
    margin-right: 20px;
  }
  .name {
    font-weight: normal;
    color: #343232;
    font-size: 17px;
  }
  p {
    opacity: 0.6;
  }
  .sizeColor {
    width: 70px;
  }

  .sizeColor + span {
    opacity: 0.6;
  }
  .mt4 {
    margin-top: 4px;
  }
}
.hideBtn{
  display: none;
}
.btnCancel {
  border: none;
  background: transparent;
}
