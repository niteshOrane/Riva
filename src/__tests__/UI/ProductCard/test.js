import React from 'react';
import renderer from 'react-test-renderer';
import ProductCard from '../../../components/common/Cards/ProductCard';

test('Product card with color functionality', () => {
  const component = renderer.create(
    <ProductCard
      product={{
        id: 1,
        src: 'assets/images/recentGirl.png',
        title: 'Contrast Paisley Printed Pleated Skirt',
        price: '75.90',
      }}
    />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
