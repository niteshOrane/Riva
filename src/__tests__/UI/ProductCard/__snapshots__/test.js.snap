// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product card with color functionality 1`] = `
<div
  className="productCard"
>
  <div
    className="imageContainer"
  >
    <img
      alt="No image available"
      onError={[Function]}
      src="assets/images/recentGirl.png"
    />
  </div>
  <div
    className="actionContainer"
  >
    <div>
      <span
        className="material-icons-outlined font-light-black"
      >
        favorite_border
      </span>
    </div>
    <div>
      <span
        className="material-icons-outlined font-light-black"
      >
        search
      </span>
    </div>
    <div>
      <span
        className="material-icons-outlined font-light-black"
      >
        shopping_cart
      </span>
    </div>
  </div>
  <div
    className="productName"
  />
  <div
    className="productPrice"
  >
    <div
      className="was"
    >
      Was 
      $
    </div>
    <div
      className="now"
    >
      Now 
      $
    </div>
  </div>
  <div
    className="productColors"
  >
    <div
      className="color color_red"
    />
    <div
      className="color color_oranage"
    />
    <div
      className="color color_blue"
    />
  </div>
</div>
`;
